iroh-relay/src/client/conn.rs

```diff
@@ -308,140 +146,113 @@ impl Stream for ConnReader {
     }
 }
 
-impl Sink<Frame> for ConnWriter {
-    type Error = std::io::Error;
+impl Sink<Frame> for Conn {
+    type Error = ConnSendError;
 
     fn poll_ready(mut self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Result<(), Self::Error>> {
         match *self {
-            Self::Derp(ref mut ws) => Pin::new(ws).poll_ready(cx),
-            Self::Ws(ref mut ws) => Pin::new(ws).poll_ready(cx).map_err(tung_wasm_to_io_err),
+            Self::Relay { ref mut conn } => Pin::new(conn).poll_ready(cx).map_err(Into::into),
+            Self::Ws { ref mut conn, .. } => Pin::new(conn)
+                .poll_ready(cx)
+                .map_err(tung_wasm_to_io_err)
+                .map_err(Into::into),
```

Comment:
You're using this a bunch.
At this point I think it makes sense to change `tung_wasm_to_io_err` into `tung_wasm_to_conn_send_err` and make it return `ConnSendError`.

Or perhaps - now that we own `ConnSendError` - we could implement `From<tungstenite_wasm::Error> for ConnSendError`.